src/jit/codegencommon.cpp

```diff
@@ -7369,10 +7398,18 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
     // Can't have a call until we have enough padding for rejit
     genPrologPadForReJit();
 
+#if !defined(UNIX_AMD64_ABI)
     // This will emit either
     // "call ip-relative 32-bit offset" or
     // "mov rax, helper addr; call rax"
     genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN);
+#else
+    // We can use any callee trash register (other than RAX, RDI, RSI) for call target.
+    // We use R11 here. This will emit either
+    // "call ip-relative 32-bit offset" or
+    // "mov r11, helper addr; call r11"
+    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN, REG_R11);
```

Comment:
@sergign60 - note that arg registers are already homed to their stack locations and they are available for use.  RCX is one of the arg registers.  In such a case we can use rcx for generating enter callback.  IMO, it is best to avoid #ifdef code or codegen differences between window vs unix amd64 as much as possible.